stacks_test.go

```diff
@@ -0,0 +1,73 @@
+package scalingo
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/Scalingo/go-scalingo/v5/http/httpmock"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+var isDeprecatedCases = map[string]struct {
+	date       time.Time
+	deprecated bool
+}{
+	"test isDeprecated is false when deprecation date is null":          {time.Time{}, false},
+	"test isDeprecated is true when deprecation date is today's date":   {time.Now(), true},
+	"test isDeprecated is true when deprecation date is in the past":    {time.Now().AddDate(0, 0, -1), true},
+	"test isDeprecated is false when deprecation date is in the future": {time.Now().AddDate(0, 0, 1), false},
+}
+
+var stacksListCases = map[string]struct {
+	json          string
+	expectedStack Stack
+}{
+	"test StacksList with stacks with deprecated_at being null": {
+		json:          `{"stacks": [{"deprecated_at": null}]}`,
+		expectedStack: Stack{DeprecatedAt: DeprecationDate{time.Time{}}},
+	},
+	"test StacksList with stacks with deprecated_at being set": {
+		json:          `{"stacks": [{"deprecated_at": "2022-08-31"}]}`,
+		expectedStack: Stack{DeprecatedAt: DeprecationDate{time.Date(2022, 8, 31, 0, 0, 0, 0, time.UTC)}},
+	},
+}
+
+func TestStackIsDeprecated(t *testing.T) {
+	for message, test := range isDeprecatedCases {
+		t.Run(message, func(t *testing.T) {
+			stack := Stack{DeprecatedAt: DeprecationDate{test.date}}
+			if stack.IsDeprecated() != test.deprecated {
+				t.Errorf("IsDeprecated expected to be %t, got %t", test.deprecated, stack.IsDeprecated())
+			}
```

Comment:
That's how we usually do:
```suggestion
assert.Equal(t, test.deprecated, stack.IsDeprecated())
```

You can refer to the other tests in the lib to check how this dependency works.

Reply:
That's much better indeed  🤩